renderdoc/driver/vulkan/vk_core.cpp

```diff
@@ -2309,6 +2309,7 @@ void WrappedVulkan::StartFrameCapture(DeviceOwnedWindow devWnd)
   // will check to see if they need to markdirty or markpendingdirty
   // and go into the frame record.
   {
+    CheckPendingCommandBufferCallbacks();
```

Comment:
I've had to move this to just outside the `SCOPED_WRITELOCK(m_CapTransitionLock)` because the readback mem clean up calls `GPUBuffer::Destroy()` which calls into `WrappedVulkan::vkDestroyBuffer` and `vkFreeMemory` - both of which call `SCOPED_READLOCK(m_CapTransitionLock)`.

On bionic this appears to have no ill effects but on glibc it seems to put the lock into a weird state where the readers are still decremented even though they though weren't incremented, as the read count is unsigned this can eventually cause a wrap around.  Then when `SCOPED_WRITELOCK(m_CapTransitionLock)` is called again in `WrappedVulkan::EndFrameCapture` the lock thinks it has readers so waits forever.

Reply:
The RW lock is not recursive I believe so that's why it will go wrong when held recursively.  Moving it here should be fine but can you move it outside of the scope just to be very clear that it's a deliberate choice and not an accidental placement.